documentation/modules/oauth/proc-oauth-authorization-keycloak-example.adoc

```diff
@@ -0,0 +1,635 @@
+----
+
+. Create a truststore for TLS connection to the Kafka brokers.
++
+[source,shell,subs="+quotes"]
+----
+keytool -keystore /tmp/truststore.p12 -storetype pkcs12 -alias sso -storepass $STOREPASS -import -file /tmp/sso.crt -noprompt
+----
+
+. Use the Kafka bootstrap address as the hostname of the Kafka broker and the `tls` listener port (9093) to prepare a certificate for the Kafka broker.
++
+[source,shell]
+----
+KAFKA_HOST_PORT=my-cluster-kafka-bootstrap:9093
+STOREPASS=storepass
+
+echo "Q" | openssl s_client -showcerts -connect $KAFKA_HOST_PORT 2>/dev/null | awk ' /BEGIN CERTIFICATE/,/END CERTIFICATE/ { print $0 } ' > /tmp/my-cluster-kafka.crt
+----
+
+. Add the certificate for the Kafka broker to the truststore.
++
+[source,shell]
+----
+keytool -keystore /tmp/truststore.p12 -storetype pkcs12 -alias my-cluster-kafka -storepass $STOREPASS -import -file /tmp/my-cluster-kafka.crt -noprompt
+----
++
+Keep the session open to check authorized access.
+
+[id='proc-oauth-authorization-keycloak-example-check_{context}']
+== Checking authorized access to Kafka using a CLI Kafka client session
+
+Check the authorization rules applied through the Keycloak realm using an interactive CLI session.
+Apply the checks using Kafka's example producer and consumer clients to create topics with user and service accounts that have different levels of access.
+
+Use the `team-a-client` and `team-b-client` clients to check the authorization rules.
+Use the `alice` admin user to perform additional administrative tasks on Kafka.
+
+The Strimzi Kafka image used in this example contains Kafka producer and consumer binaries.
+
+.Prerequisites
+
+* ZooKeeper and Kafka are running in the Kubernetes cluster to be able to send and receive messages.
+* The xref:proc-oauth-authorization-keycloak-example-authentication_{context}[interactive CLI Kafka client session] is started.
++
+{ApacheKafkaDownload}.
+
+.Setting up client and admin user configuration
+
+. Prepare a Kafka configuration file with authentication properties for the `team-a-client` client.
++
+[source,shell,subs="+quotes"]
+----
+SSO_HOST=_SSO-HOSTNAME_
+
+cat > /tmp/team-a-client.properties << EOF
+security.protocol=SASL_SSL
+ssl.truststore.location=/tmp/truststore.p12
+ssl.truststore.password=$STOREPASS
+ssl.truststore.type=PKCS12
+sasl.mechanism=OAUTHBEARER
+sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required \
+  oauth.client.id="team-a-client" \
+  oauth.client.secret="team-a-client-secret" \
+  oauth.ssl.truststore.location="/tmp/truststore.p12" \
+  oauth.ssl.truststore.password="$STOREPASS" \
+  oauth.ssl.truststore.type="PKCS12" \
+  oauth.token.endpoint.uri="https://$SSO_HOST/auth/realms/kafka-authz/protocol/openid-connect/token" ;
+sasl.login.callback.handler.class=io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler
+EOF
+----
++
+The SASL/OAUTHBEARER mechanism is used.
```

Comment:
Referred to elsewhere as the OAUTHBEARER mechanism (although there are some exceptions).